test/e2e/network/network_policy.go

```diff
@@ -1874,87 +1874,115 @@ var _ = SIGDescribe("NetworkPolicy [Feature:SCTPConnectivity][LinuxOnly][Disrupt
-				testCanConnectProtocol(f, nsB, "client-a", service, 80, v1.ProtocolSCTP)
+				testCanConnectProtocol(f, nsB, "client-a", service, 80, v1.ProtocolSCTP, 3)
 			})
 			ginkgo.By("Creating client-b, not in server's namespace, which should not be able to contact the server.", func() {
-				testCannotConnectProtocol(f, nsB, "client-b", service, 80, v1.ProtocolSCTP)
+				testCannotConnectProtocol(f, nsB, "client-b", service, 80, v1.ProtocolSCTP, 3)
 			})
 		})
 	})
 })
 
-func testCanConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCanConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+func testCanConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, tryCount int) {
+	testCanConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP, tryCount)
 }
 
-func testCannotConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCannotConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+func testCannotConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, tryCount int) {
+	testCannotConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP, tryCount)
 }
 
-func testCanConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol) {
-	ginkgo.By(fmt.Sprintf("Creating client pod %s that should successfully connect to %s.", podName, service.Name))
-	podClient := createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
-	defer func() {
-		ginkgo.By(fmt.Sprintf("Cleaning up the pod %s", podClient.Name))
-		if err := f.ClientSet.CoreV1().Pods(ns.Name).Delete(context.TODO(), podClient.Name, metav1.DeleteOptions{}); err != nil {
-			framework.Failf("unable to cleanup pod %v: %v", podClient.Name, err)
+func testCanConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol, tryCount int) {
+	i := 0
+	keepTrying := true
+	var podClient *v1.Pod
+	var conErr error
+
+	// Network policy can take some time to program, especially in scaled/busy environments. Let's allow some retries to
+	// account for that, until we have status field available on network policy object
+	for i < tryCount && keepTrying {
+		ginkgo.By(fmt.Sprintf("Creating client pod %s that should successfully connect to %s.", podName, service.Name))
+		framework.Logf(fmt.Sprintf("Count: %d of tryCount: %d - Creating client pod %s that should successfully connect to %s.", i+1, tryCount, podName, service.Name))
+		podClient = createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
+		conErr = checkConnectivity(f, ns, podClient, service)
+		if conErr == nil {
+			// got the expected result, stop trying
+			keepTrying = false
```

Comment:
you don't need `keepTrying`. Just `break`

Reply:
Done. Also, cleaned up the for loop iterator i var.